src/zen/websocket_server.cpp

```diff
@@ -0,0 +1,1360 @@
+        return OK;
+    }
+
+    int sendRawMempool(const std::string& clientRequestId) {
+        LOCK(mempool.cs);
+
+        std::list<uint256> hashes;
+        BOOST_FOREACH(const PAIRTYPE(uint256, CTxMemPoolEntry)& entry, mempool.mapTx)
+        {
+           hashes.push_back(entry.first);
+        }
+        sendMempool(hashes.size(), hashes, WsEvent::MSG_RESPONSE, clientRequestId);
+        return OK;
+
+    }
+
+    int sendMempoolTxs(const UniValue& hashes, const std::string& clientRequestId) {
+        LOCK(mempool.cs);
+
+        std::list<std::string> txs;
+        for(const UniValue& hash : hashes.getValues()) {
+            if (!hash.isStr())
+            {
+                LogPrint("ws", "%s():%d - invalid string\n", __func__, __LINE__);
+                return INVALID_PARAMETER;
+            }
+			uint256 txid(uint256S(hash.get_str()));
+			auto entry = mempool.mapTx.find(txid);
+			if (entry != mempool.mapTx.end()) {
+				txs.push_back(EncodeHexTx(
+						static_cast<CTransaction>(entry->second.GetTx())));
+			} else {
+				LogPrint("ws", "%s():%d - mempool transaction not found!\n", txid.ToString(), __LINE__);
+			}
+
+        }
+        sendTxs(txs, WsEvent::MSG_RESPONSE, clientRequestId);
+        return OK;
+    }
+
+    int sendNodeStaticConfiguration(const std::string& clientRequestId) {
+        int version = CLIENT_VERSION;
+        int protocolVersion = PROTOCOL_VERSION;
+        int timeoffset = 0;
+        int relayfee = minRelayTxFee.GetFeePerK();
+        proxyType p;
+        GetProxy(NET_IPV4, p);
+        std::string proxy = p.IsValid() ? p.proxy.ToStringIPPort() : std::string();
+        std::string errors = GetWarnings("statusbar");
+        int blocks = (int)chainActive.Height();
+        int connections = (int)vNodes.size();
+        bool testnet = Params().TestnetToBeDeprecatedFieldRPC();
+        double difficulty = (double)GetDifficulty();
+        double networkDifficulty = (double)GetNetworkDifficulty();
+		std::list<std::string> peers;
+        {
+			LOCK(cs_vNodes);
+			BOOST_FOREACH(CNode* pnode, vNodes) {
+				CNodeStats stats;
+				pnode->copyStats(stats);
+				peers.push_back(stats.addrName);
+			}
+        }
+        sendNodeConf(version, protocolVersion, timeoffset, relayfee, proxy, errors, blocks, connections, testnet, difficulty, networkDifficulty,
+            peers, WsEvent::MSG_RESPONSE, clientRequestId);
+        return OK;
+    }
+
+    int sendEstimateFee(std::string strBlock, const std::string& clientRequestId) {
+        int nBlocks = -1;
+        try {
+            nBlocks = std::stoi(strBlock);
+        } catch (const std::exception &e) {
+            LogPrint("ws", "%s():%d - %s\n", __func__, __LINE__, e.what());
+            return INVALID_PARAMETER;
+        }
+
+        if (nBlocks < 1) {
+            nBlocks = 1;
+        }
+
+        CFeeRate feeRate = mempool.estimateFee(nBlocks);
```

Comment:
verify if lock is needed

Reply:
The RPC command where the implementation is taken doesn't use any lock. (https://github.com/HorizenOfficial/zen/blob/master/src/rpc/mining.cpp#L786). However this event is never called from the explorer, maybe i can delete it.